BSP/Simulator/Src/BSP_UART.c

```diff
@@ -121,105 +41,3 @@ uint32_t BSP_UART_ReadLine(char *str, UART_Port usart) {
 uint32_t BSP_UART_Write(char *str, uint32_t len, UART_Port usart) {
     return printf("%.*s", len, str);
```

Comment:
the test file for this doesnt really exist, not like we really need it but here is the updated test code for `Test_BSP_UART.c`, Im too lazy to do a push for it 

#include "common.h"
#include "config.h"
#include "BSP_UART.h"
#include "BSP_PLL.h"
#ifndef SIMULATION
#include "stm32f4xx.h"
#endif 
void foo(void){
    return;
}

int main() {
    BSP_UART_Init(foo, foo, UART_USB);
    char* str = (char*) malloc(20*sizeof(char));
    while(1) {
        int num = BSP_UART_ReadLine(str, UART_USB);
        if(num > 0) {
            printf("%s\n\r", str);
        }
    }
}

Reply:
I do not think we need to redo the BSP files to accommodate for testing Simulation functionality. We would mainly be using it to test hardware (since these tests have worked before with the same code, they should work again if we change the hardware and the hardware works correctly).